File: packages/javascript/lib/word/randFurniture.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Furniture and the furnishings of a home.
 *
 * `randWord({ theme: 'furniture' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randFurniture({ language: 'ko', count: 3 }); // ['흔들의자', '요람', '책장']
 * randFurniture({ language: 'en', count: 3 }); // ['Hammock', 'Cradle', 'Wardrobe']
 */
export function randFurniture(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Furniture words with the language and theme each one came from.
 *
 * @example
 * randFurniture({ language: 'ko', output: 'detail' });
 * // [{ word: '흔들의자', language: 'ko', theme: 'furniture' }]
 */
export function randFurniture(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randFurniture(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('furniture', options);
}

File: packages/dart/lib/src/word/rand_furniture.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Furniture and the furnishings of a home.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.furniture] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randFurniture(language: WordLanguage.ko, count: 3); // [흔들의자, 요람, 책장]
/// randFurniture(language: WordLanguage.en, count: 3); // [Hammock, Cradle, Wardrobe]
/// ```
List<String> randFurniture({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.furniture,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);
